### ISSUE TITLE
How to add a command for the scripts for a new project via the template file?

### ISSUE THREAD

**Issue by XingXiaoWu** [react-native-community/cli#392]
I want to create my own template, and so far I'm about halfway there.

My file structure is as follows

<img width="438" alt="WeChat43f4293b0829b68fb1b66d06d7d4d087" src="https://user-images.githubusercontent.com/22769131/57903633-6faa2480-78a1-11e9-8773-c473119d88f6.png">

I created dependencies.json, devDependencies.json was successfully added to package.json, index.js, babel.config.js was also successfully replaced.

But scripts.json didn't work. The script in the new project's package.json didn't add the commands I wrote in scripts.json. I tried to write scripts.json as scripts.js and it didn't take effect.

**My question is, how do I add a new command to the scripts in the package.json of a project created via template?**

**Comment by thymikee** [comment]
Yea, old way to init is pretty daunting and limited and AFAIK it's not possible because the `react-native-cli` is creating this package.json.

However, we've been working on the new `init` that doesn't have such limitations and you can create whichever template you like. Here are the docs on that: https://github.com/react-native-community/cli/blob/master/docs/init.md

You'll notice it's available since not-yet-present RN 0.60. That's true. However you can use this CLI as standalone like, where init command is available in `next` channel (currently having `v2.0.0-alpha.20` there):

```
npx @react-native-community/cli@next init ProjectName --template your-template
```

I'm going to close it, as old init is in a maintenance mode and we're not going to change things.

**Comment by XingXiaoWu** [comment]
Sorry, I have read this document, but it doesn't solve my problem

My question is, how do I add a new command to the scripts in the package.json of a project created via template?

**Comment by thymikee** [comment]
How do you run your init command? 

From your screenshot it looks like you use:
1) `template.config.js` – this is only understood by new init
2) `dependencies.json` – this is irrelevant for new init, used by old init
3) `devDependencies.json` – same ^
4) `scripts.json` – same ^

In new init you just create your `package.json` with anything inside (i.e. certain scripts added) and it will be copied as is. It's as simple as that.

**Comment by XingXiaoWu** [comment]
I think I probably understand what you mean, but I am confused and hope to get your reply.
1. From the point of view, this is prepared for react-native 0.60.0. The current latest version is 0.59.8. Is it applicable?
2. In the new init, I don't want to destroy the contents of the original package.json. I just want to add some configuration I need. I don't know if the content I created in package.json will be overwritten in the new init. Add after comparison?

Thank you very much for your reply

**Comment by thymikee** [comment]
You can use the new init through standalone CLI to install _any_ version of React Native, it will just be available in `react-native` from 0.60 (using `npx react-native init` spawn new init, and globally installed `react-native init` will use old for backcompat)

See the example of new template here: https://github.com/Esemesek/react-native-new-template/

Feel free to compare the `template/package.json` to `YourProject/package.json` – the only difference will be the "name" field. Everything else, not related to your project name, will be copied and preserved. So if your `template/package.json` have "scripts.doStuff": "stuff" it's gonna still be there. Nothing is destroyed. I wonder where do you get that impression that something is? Please just try it and see for your self.

**Comment by XingXiaoWu** [comment]
I think I probably understand what you mean,let me try,thanks:)
